src/FSharp.Formatting.Common/YaafFSharpScripting.fs

```diff
@@ -464,26 +463,29 @@ module internal ArgParser =
         else
             None
 
+    [<return: Struct>]
     let (|FsiBoolArg|_|) argName s =
         match s with
         | StartsWith argName rest ->
             if String.IsNullOrWhiteSpace rest then
-                Some true
+                ValueSome true
             else
                 match rest with
-                | "+" -> Some true
-                | "-" -> Some false
-                | _ -> None
-        | _ -> None
+                | "+" -> ValueSome true
+                | "-" -> ValueSome false
+                | _ -> ValueNone
+        | _ -> ValueNone
 
 open ArgParser
 
+[<Struct>]
```

Comment:
The heuristic would be here to add `[<Struct>]` when the DU only has cases without any fields?

Reply:
Yes, Enums and similar are struct per default, so if you do this:
```fsharp
type MyE =
| A = 1
| B = 2
```
it's fine as is. But if you do this:
```
type MyE =
| A 
| B 
``
...it's not a struct. So basically it'll generate a class of MyE containing a class of A and class of B and an instance of either.